nni/tools/nnictl/config_schema.py

```diff
@@ -364,7 +370,7 @@ def validate(self, data):
 frameworkcontroller_trial_schema = {
     'trial': {
         'codeDir':  setPathCheck('codeDir'),
-        'taskRoles': [{
+        Optional('taskRoles'): [{
```

Comment:
If set `taskRoles` as optional field, better to add validation, refer https://github.com/microsoft/nni/blob/master/nni/tools/nnictl/config_schema.py#L536

Reply:
I totally missed that validation part of the code! Super cool. I added a (even a little more extended) verification for the frameworkcontroller job config.